lint/param-name_checker.go

```diff
@@ -0,0 +1,74 @@
+package lint
+
+import (
+	"fmt"
+	"go/ast"
+	"strings"
+)
+
+// ParamNameChecker detects potential issues in function parameter names.
+//
+// Rationale: makes godoc output better.
+type ParamNameChecker struct {
+	ctx *Context
+
+	loudNames map[string]bool
+
+	warnings []Warning
+}
+
+// NewParamNameChecker returns initialized checker for Go functions param names.
+func NewParamNameChecker(ctx *Context) *ParamNameChecker {
+	return &ParamNameChecker{
+		ctx: ctx,
+		loudNames: map[string]bool{
+			"IN":    true,
+			"OUT":   true,
+			"INOUT": true,
+		},
+	}
+}
+
+// Check runs parameter names checks for f.
+//
+// Features
+//
+// 1. Detects somewhat common "loud" identifiers, like IN or OUT.
+//    Suggests to replace them with down-case versions (in, out).
+//
+// 2. If capitalized name is not recognized as "loud",
+//    treat it as "redundantly exported".
+//    Suggests to use non-capitalized identifier.
+func (c *ParamNameChecker) Check(f *ast.File) []Warning {
+	c.warnings = c.warnings[:0]
+	for _, decl := range collectFuncDecls(f) {
+		for _, param := range collectFuncParams(decl) {
+			for _, id := range param.Names {
+				switch {
+				case c.loudNames[id.Name]:
+					c.warnLoud(id)
```

Comment:
Maybe we need also add camel case check here?

Reply:
`golint` does that.